src/backend/gl/src/command.rs

```diff
@@ -628,41 +631,33 @@ impl command::RawCommandBuffer<Backend> for RawCommandBuffer {
         //  >= GL 4.5: Invalidate framebuffer attachment when store op is `DONT_CARE`.
 
         // 2./3.
-        self.push_cmd(Command::BindFrameBuffer(
-            glow::DRAW_FRAMEBUFFER,
-            *framebuffer,
-        ));
-
         let mut clear_values_iter = clear_values.into_iter();
         let attachment_clears = render_pass
             .attachments
             .iter()
             .enumerate()
-            .map(|(i, attachment)| {
+            .map(|(id, attachment)| {
                 let cv = if attachment.has_clears() {
-                    Some(clear_values_iter.next().unwrap())
+                    clear_values_iter.next().unwrap()
                 } else {
-                    None
+                    return None;
                 };
-                AttachmentClear {
-                    subpass_id: render_pass.subpasses.iter().position(|sp| sp.is_using(i)),
-                    value: if attachment.ops.load == pass::AttachmentLoadOp::Clear {
-                        Some(*cv.as_ref().unwrap().borrow())
-                    } else {
-                        None
-                    },
-                    stencil_value: if attachment.stencil_ops.load == pass::AttachmentLoadOp::Clear {
-                        Some(cv.unwrap().borrow().depth_stencil.stencil)
-                    } else {
-                        None
-                    },
-                }
+
+                let (subpass, index) = render_pass.subpasses.iter().enumerate().filter_map(|(i, sp)| {
+                    let index = sp.attachment_using(id)?;
+                    Some((i, index))
+                }).next()?;
```

Comment:
this looks weird to me: so we are constructing an iterator from `subpasses` and only checking the first element of it?

Reply:
Exactly.

Reply:
I suppose it would look cleaner with `iter().position()` instead, but it hardly matters :)

Reply:
It was cleaner with `.position()`, but I need not only position now.